demos/projects/ST/stm32h745i_discovery/cm7/main.c

```diff
@@ -0,0 +1,676 @@
+ *
+ * Called if a call to pvPortMalloc() fails because there is insufficient
+ * free memory available in the FreeRTOS heap.  pvPortMalloc() is called
+ * internally by FreeRTOS API functions that create tasks, queues, software
+ * timers, and semaphores.  The size of the FreeRTOS heap is set by the
+ * configTOTAL_HEAP_SIZE configuration constant in FreeRTOSConfig.h.
+ *
+ */
+void vApplicationMallocFailedHook()
+{
+    taskDISABLE_INTERRUPTS();
+
+    for( ; ; )
+    {
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Loop forever if stack overflow is detected.
+ *
+ * If configCHECK_FOR_STACK_OVERFLOW is set to 1,
+ * this hook provides a location for applications to
+ * define a response to a stack overflow.
+ *
+ * Use this hook to help identify that a stack overflow
+ * has occurred.
+ *
+ */
+void vApplicationStackOverflowHook( TaskHandle_t xTask,
+                                    char * pcTaskName )
+{
+    portDISABLE_INTERRUPTS();
+
+    /* Loop forever */
+    for( ; ; )
+    {
+    }
+}
+/*-----------------------------------------------------------*/
+
+void vApplicationIdleHook( void )
+{
+}
+
+/*-----------------------------------------------------------*/
+
+void prvGetRegistersFromStack( uint32_t * pulFaultStackAddress )
+{
+/* These are volatile to try and prevent the compiler/linker optimising them
+ * away as the variables never actually get used.  If the debugger won't show the
+ * values of the variables, make them global my moving their declaration outside
+ * of this function. */
+    volatile uint32_t r0;
+    volatile uint32_t r1;
+    volatile uint32_t r2;
+    volatile uint32_t r3;
+    volatile uint32_t r12;
+    volatile uint32_t lr;  /* Link register. */
+    volatile uint32_t pc;  /* Program counter. */
+    volatile uint32_t psr; /* Program status register. */
+
+    r0 = pulFaultStackAddress[ 0 ];
+    r1 = pulFaultStackAddress[ 1 ];
+    r2 = pulFaultStackAddress[ 2 ];
+    r3 = pulFaultStackAddress[ 3 ];
+
+    r12 = pulFaultStackAddress[ 4 ];
+    lr = pulFaultStackAddress[ 5 ];
+    pc = pulFaultStackAddress[ 6 ];
+    psr = pulFaultStackAddress[ 7 ];
+
+    /* Remove compiler warnings about the variables not being used. */
+    ( void ) r0;
+    ( void ) r1;
+    ( void ) r2;
+    ( void ) r3;
+    ( void ) r12;
+    ( void ) lr;  /* Link register. */
+    ( void ) pc;  /* Program counter. */
+    ( void ) psr; /* Program status register. */
+
+    /* When the following line is hit, the variables contain the register values. */
+    for( ; ; )
+    {
+    }
+}
+/*-----------------------------------------------------------*/
+
+/* Psuedo random number generator.  Just used by demos so does not need to be
+ * secure.  Do not use the standard C library rand() function as it can cause
+ * unexpected behaviour, such as calls to malloc(). */
+int iMainRand32( void )
+{
+    static UBaseType_t uxlNextRand; /*_RB_ Not seeded. */
```

Comment:
Call https://github.com/Azure-Samples/iot-middleware-freertos-samples/pull/49/files#diff-728316c27f5d00d5c0db94144f86ec69d2a700015fad0dca72a7930a0c34fb08R119 instead?

Reply:
I am not able to follow this link ... Could please provide the link to specific line?

Reply:
Sorry - it was the already implemented secure random function instead of implementing a pseudo-random function - likely wrong paste.